sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlockBlobAsyncApiTests.java

```diff
@@ -0,0 +1,2658 @@
+        Mono<BlockBlobItem> uploadOperation = blobAsyncClient.upload(
+            Flux.fromIterable(dataList).publish().autoConnect(),
+            new ParallelTransferOptions().setMaxSingleUploadSizeLong(4L * Constants.MB), true);
+
+        StepVerifier.create(uploadOperation.then(collectBytesInBuffer(blockBlobAsyncClient.downloadStream())))
+            .assertNext(it -> assertTrue(compareListToBuffer(dataList, it)))
+            .verifyComplete();
+
+        StepVerifier.create(blockBlobAsyncClient.listBlocks(BlockListType.ALL))
+            .assertNext(it -> assertEquals(it.getCommittedBlocks().size(), blockCount))
+            .verifyComplete();
+    }
+
+    @ParameterizedTest
+    @MethodSource("bufferedUploadHandlePathingHotFluxWithTransientFailureSupplier")
+    @EnabledIf("com.azure.storage.blob.BlobTestBase#isLiveMode")
+    public void bufferedUploadHandlePathingHotFluxWithTransientFailure(int[] dataSizeList, int blockCount) {
+        BlobAsyncClient clientWithFailure = getBlobAsyncClient(
+            ENVIRONMENT.getPrimaryAccount().getCredential(),
+            blobAsyncClient.getBlobUrl(),
+            new TransientFailureInjectingHttpPipelinePolicy()
+        );
+
+        List<ByteBuffer> dataList = new ArrayList<>();
+        for (int size : dataSizeList) {
+            dataList.add(getRandomData(size));
+        }
+        Mono<BlockBlobItem> uploadOperation = clientWithFailure.upload(
+            Flux.fromIterable(dataList).publish().autoConnect(),
+            new ParallelTransferOptions().setMaxSingleUploadSizeLong(4L * Constants.MB), true);
+
+        StepVerifier.create(uploadOperation.then(collectBytesInBuffer(blockBlobAsyncClient.downloadStream())))
+            .assertNext(it -> assertTrue(compareListToBuffer(dataList, it)))
+            .verifyComplete();
+
+        StepVerifier.create(blockBlobAsyncClient.listBlocks(BlockListType.ALL))
+            .assertNext(it -> assertEquals(it.getCommittedBlocks().size(), blockCount))
+            .verifyComplete();
+    }
+
+    private static Stream<Arguments> bufferedUploadHandlePathingHotFluxWithTransientFailureSupplier() {
+        return Stream.of(
+            Arguments.of(new int[]{10, 100, 1000, 10000}, 0),
+            Arguments.of(new int[]{4 * Constants.MB + 1, 10}, 2),
+            Arguments.of(new int[]{4 * Constants.MB, 4 * Constants.MB}, 2));
+    }
+
+    @Test
+    public void bufferedUploadIllegalArgumentsNull() {
+        StepVerifier.create(blobAsyncClient.upload(null, new ParallelTransferOptions()
+                .setBlockSizeLong(4L)
+                .setMaxConcurrency(4), true))
+            .verifyErrorSatisfies(it -> assertInstanceOf(NullPointerException.class, it));
+    }
+
+    @ParameterizedTest
+    @MethodSource("bufferedUploadIllegalArgsOutOfBoundsSupplier")
+    public void bufferedUploadIllegalArgsOutOfBounds(long bufferSize, int numBuffs) {
+        assertThrows(IllegalArgumentException.class, () -> new ParallelTransferOptions()
```

Comment:
Should think about moving this test out of here as this should be in a test class specific to ParallelTransferOptions that isn't recorded.